ublox/src/ubx_packets/packets.rs

```diff
@@ -1981,11 +1983,68 @@ struct EsfMeas {
     calib_tag: u32,
 }
 
-// #[ubx_packet_recv]
-// #[ubx(class = 0x10, id = 0x03, fixed_payload_len = 16)]
-// struct EsfRaw {
-//     msss: u32,
-// }
+#[ubx_packet_recv]
+#[ubx(class = 0x10, id = 0x03, max_payload_len = 1240)]
+struct EsfRaw {
+    msss: u32,
+    #[ubx(map_type = EsfRawIter,
+    from = esf_raw::convert_to_iter,
+    is_valid = esf_raw::is_valid)]
+    iter: [u8; 0],
+}
+
+pub struct EsfRawIter<'a> {
+    bytes: &'a [u8],
+    offset: usize,
+}
+
+pub struct EsfRawInfo {
+    data: u32,
+    sensor_time_tag: u32,
+}
+
+impl<'a> core::iter::Iterator for EsfRawIter<'a> {
+    type Item = EsfRawInfo;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset < self.bytes.len() {
+            let data_slice: [u8; 4] = (&self.bytes[self.offset..self.offset + 4])
```

Comment:
should probably break this out into a helper function or something

Reply:
I think with a little tweaking it's probably clean enough without a helper 

```rust
    fn next(&mut self) -> Option<Self::Item> {
        if self.offset >= self.bytes.len() {
            return None;
        }
        let data_slice: [u8; 4] = (&self.bytes[self.offset..self.offset + 4])
            .try_into()
            .unwrap();
        let ss_slice: [u8; 4] = (&self.bytes[self.offset + 4..self.offset + 8])
            .try_into()
            .unwrap();
        self.offset += 8;
        Some(EsfRawInfo {
            data: u32::from_ne_bytes(data_slice),
            sensor_time_tag: u32::from_ne_bytes(ss_slice),
        })
    }
```

also does the offset check need to be `self.offset + 8 < self.bytes.len()` or  `self.offset + 8 >= self.bytes.len()` to ensure there are enough bytes for the next item?

Reply:
Also why `from_ne_bytes` is that what the spec says? I'd assume that it should be a fixed endianness not native (and most likely little endian)

Reply:
I think it just need to be `self.offset < self.bytes.len()` since we know by the `is_valid` method it's multiple of 8 but can be more safe/readable to do `self.offset + 8 <= self.bytes.len()` (need that `=` to align the end of the offset)

Also yea oops its `from_le_bytes`

Reply:
Ahh right that makes sense